tracer/test/Datadog.Trace.Tests/Debugger/ConcurrentAdaptiveCacheTests.cs

```diff
@@ -0,0 +1,299 @@
+
+            // Add items with short expiration
+            for (int i = 0; i < 50; i++)
+            {
+                cache.Add(i, TimeSpan.FromMilliseconds(50), keys: $"key{i}");
+            }
+
+            // Wait for items to expire
+            await Task.Delay(100);
+
+            // Manually trigger cleanup
+            int itemsRemoved = cache.PerformCleanup();
+            Assert.True(itemsRemoved > 0);
+
+            // Check that cleanup interval decreased
+            var initialInterval = cache.CurrentCleanupInterval;
+            cache.AdjustCleanupInterval(itemsRemoved);
+            Assert.True(cache.CurrentCleanupInterval < initialInterval);
+
+            // Add more items with longer expiration
+            for (int i = 50; i < 100; i++)
+            {
+                cache.Add(i, TimeSpan.FromSeconds(10), keys: $"key{i}");
+            }
+
+            // Manually trigger cleanup again
+            itemsRemoved = cache.PerformCleanup();
+            Assert.True(itemsRemoved == 0);
+
+            // Check that cleanup interval increased
+            initialInterval = cache.CurrentCleanupInterval;
+            cache.AdjustCleanupInterval(itemsRemoved);
+            Assert.True(cache.CurrentCleanupInterval > initialInterval);
+        }
+
+        [Fact]
+        public void Adaptive_Cleanup_Should_Respect_Min_And_Max_Intervals()
+        {
+            var cache = new ConcurrentAdaptiveCache<string, int>(capacity: 100);
+
+            // Simulate many cleanups with no items removed
+            for (int i = 0; i < 100; i++)
+            {
+                cache.AdjustCleanupInterval(0);
+            }
+
+            Assert.True(cache.CurrentCleanupInterval.TotalSeconds <= ConcurrentAdaptiveCache<string, int>.MaxCleanupIntervalSeconds);
+
+            // Simulate many cleanups with items removed
+            for (int i = 0; i < 100; i++)
+            {
+                cache.AdjustCleanupInterval(10);
+            }
+
+            Assert.True(cache.CurrentCleanupInterval.TotalSeconds >= ConcurrentAdaptiveCache<string, int>.MinCleanupIntervalSeconds);
+        }
+
+        [Fact]
+        public async Task AdaptiveCleanupAsync_Should_Continuously_Adjust_Interval()
+        {
+            var cache = new ConcurrentAdaptiveCache<string, int>(capacity: 100);
+            var initialInterval = cache.CurrentCleanupInterval;
+
+            // Start the cleanup task
+            var cleanupTask = cache.AdaptiveCleanupAsync();
+
+            // Add items with varying expiration times
+            for (int i = 0; i < 50; i++)
+            {
+                cache.Add(i, TimeSpan.FromMilliseconds(50), keys: $"shortKey{i}");
+            }
+            for (int i = 50; i < 100; i++)
+            {
+                cache.Add(i, TimeSpan.FromSeconds(10), keys: $"longKey{i}");
+            }
+
+            // Wait for some cleanups to occur
+            await Task.Delay(TimeSpan.FromSeconds(15));
```

Comment:
This is kind of a long delay for a unit-test, are you able to shorten this a little?

Reply:
Thank you for noticing! fixed in 64548d59d44e2ab9d1f8d7517841eff14b91ed18